src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -140,18 +145,36 @@ public void bind(final SparkSwagger restApi) {
     }
 
     /**
-     * HTTP endpoint to get multiple entities.
+     * HTTP endpoint to get multiple entities based on the user permissions
      */
+    // FIXME Maybe better if the user check (and filtering) was done in a pre hook?
+    // FIXME Will require further granularity for admin
     private List<T> getMany(Request req, Response res) {
-        return persistence.getAll();
+
+        Auth0UserProfile requestingUser = getUserFromRequest(req);
+        if (isUserAdmin(requestingUser)) {
+            return persistence.getAll();
+        } else if (persistence.clazz == OtpUser.class) { 
+            Bson filter = Filters.eq("_id", requestingUser.otpUser.id);
+            return persistence.getFiltered(filter);
+        } else {
+            // FIXME assumes all non admin user collections will have a user id field
```

Comment:
This seems like a reasonable design choice to me so I'd be fine with removing the FIXME.

Reply:
Updated.